liquibase-couchbase/src/test/java/liquibase/ext/couchbase/provider/ContextServiceProviderTest.java

```diff
@@ -0,0 +1,301 @@
+        when(scope.collection(collectionName)).thenReturn(collection);
+        when(collectionManager.getAllScopes()).thenReturn(scopeSpecList);
+        when(queryIndexManager.getAllIndexes(eq(SERVICE_BUCKET_NAME), any())).thenReturn(queryIndexList);
+
+        assertThat(contextServiceProvider.getScopeOfCollection(collectionName)).isEqualTo(scope);
+    }
+
+    @Test
+    void Should_return_service_collection_if_bucket_and_scope_and_collection_exist() {
+        ContextServiceProvider contextServiceProvider = new ContextServiceProvider(couchbaseLiquibaseDatabase);
+
+        String collectionName = "collection";
+
+        List<ScopeSpec> scopeSpecList = createScopeSpecs(DEFAULT_SERVICE_SCOPE);
+        scopeSpecList.forEach(it -> {
+            String scopeName = it.name();
+            Set<CollectionSpec> collectionSpecSet = createCollectionSpecs(scopeName, collectionName);
+            when(it.collections()).thenReturn(collectionSpecSet);
+        });
+        List<QueryIndex> queryIndexList = createQueryIndexes(collectionName);
+        when(couchbaseConnection.getDatabase()).thenReturn(bucket);
+        when(bucket.name()).thenReturn(SERVICE_BUCKET_NAME);
+        when(cluster.bucket(SERVICE_BUCKET_NAME)).thenReturn(bucket);
+        when(cluster.buckets()).thenReturn(bucketManager);
+        when(bucket.scope(DEFAULT_SERVICE_SCOPE)).thenReturn(scope);
+        when(scope.collection(collectionName)).thenReturn(collection);
+        when(collectionManager.getAllScopes()).thenReturn(scopeSpecList);
+        when(queryIndexManager.getAllIndexes(eq(SERVICE_BUCKET_NAME), any())).thenReturn(queryIndexList);
+
+        assertThat(contextServiceProvider.getServiceCollection(collectionName)).isEqualTo(collection);
+
+        verify(bucket).scope(DEFAULT_SERVICE_SCOPE);
+        verify(scope).collection(collectionName);
+
+        verify(bucketManager, never()).createBucket(any());
+        verify(collectionManager, never()).createCollection(any());
+        verify(collectionManager, never()).createScope(any());
+        verify(queryIndexManager, never()).createPrimaryIndex(any(), any());
+    }
+
+    @Test
+    void Should_return_service_collection_if_bucket_and_scope_and_collection_exist_but_database_missing() {
+        ContextServiceProvider contextServiceProvider = new ContextServiceProvider(couchbaseLiquibaseDatabase);
+
+        String collectionName = "collection";
+
+        List<ScopeSpec> scopeSpecList = createScopeSpecs(DEFAULT_SERVICE_SCOPE);
+        scopeSpecList.forEach(it -> {
+            String scopeName = it.name();
+            Set<CollectionSpec> collectionSpecSet = createCollectionSpecs(scopeName, collectionName);
+            when(it.collections()).thenReturn(collectionSpecSet);
+        });
+        List<QueryIndex> queryIndexList = createQueryIndexes(collectionName);
+        when(bucket.name()).thenReturn(SERVICE_BUCKET_NAME);
+        when(cluster.bucket(SERVICE_BUCKET_NAME)).thenReturn(bucket);
```

Comment:
A saw at least this line duplication, could these mocks be extracted to the separate method or partly extracted to `setUp()` method?

Reply:
It is possible only with Strictness.LENIENT. That means that this is a conditional configuration

Reply:
Please, check once again. I refactored mocking a lot